dvc/exceptions.py

```diff
@@ -268,8 +268,11 @@ def __init__(self, target_infos, stats=None):
         self.stats = stats
         targets = [str(t) for t in target_infos]
         m = (
-            "Checkout failed for following targets:\n{}\nDid you "
-            "forget to fetch?".format("\n".join(targets))
+            "Checkout failed for following targets:\n{}\nIs your "
+            "cache up to date?\n{}".format(
```

Comment:
The cache question seems unclear, how about just:
```suggestion
            "Checkout failed for following targets:\n{}\n"
            "Visit {} for more info.".format(
```
?

Reply:
"Visit {} for more info." sounds like holding a user hostage. We do need better exception message, which can be done by making `_fetch` and `_checkout` collaborate with each other, but making `CheckoutError` message generic is not a solution, it only adds confusion.

Reply:
@skshetry Good point. Maybe for a lack of better message (because a more advanced approach is needed), let's just append "Vinit {} for more info"(or someting like that) to the old message? Just in the interest of time, since the proper-proper solution will require quite a bit more effort.

Reply:
@efiop I was sort of thinking the same thing myself. I updated it to that and removed the "Fixes #4316" since that issue goes deeper than this. I think other changes are unrelated to this PR at this point?

Reply:
@efiop, yeah, that sounds good. But let's keep the issue open.